packages/datetime/test/timePickerTests.tsx

```diff
@@ -209,6 +209,214 @@ describe("<TimePicker>", () => {
+    describe("Time range - minTime and maxTime props", () => {
+        it("if defaultValue is smaller minTime, minTime becomes initial time", () => {
+            renderTimePicker({
+                defaultValue: new Date(1995, 6, 30, 12, 30),
+                minTime: new Date(1995, 6, 30, 15, 30),
+                precision: TimePickerPrecision.MILLISECOND,
+            });
+
+            const hourInput = findInputElement(Classes.TIMEPICKER_HOUR);
+
+            assert.strictEqual(hourInput.value, "15");
+            assert.strictEqual(timePicker.state.value.getHours(), 15);
+        });
+
+        it("if defaultValue is is greater than maxTime, minTime becomes initial time", () => {
+            renderTimePicker({
+                defaultValue: new Date(1995, 6, 30, 20, 30),
+                maxTime: new Date(1995, 6, 30, 18, 30),
+                minTime: new Date(1995, 6, 30, 15, 30),
+                precision: TimePickerPrecision.MILLISECOND,
+            });
+
+            const hourInput = findInputElement(Classes.TIMEPICKER_HOUR);
+
+            assert.strictEqual(hourInput.value, "15");
+            assert.strictEqual(timePicker.state.value.getHours(), 15);
+        });
+
+        it("by default, I can select any time", () => {
+            renderTimePicker({
+                precision: TimePickerPrecision.MILLISECOND,
+            });
+
+            const hourInput = findInputElement(Classes.TIMEPICKER_HOUR);
+            const minuteInput = findInputElement(Classes.TIMEPICKER_MINUTE);
+            const secondInput = findInputElement(Classes.TIMEPICKER_SECOND);
+            const millisecondInput = findInputElement(Classes.TIMEPICKER_MILLISECOND);
+
+            /* I can select default minTime */
+            changeInputThenBlur(hourInput, "0");
+            assert.strictEqual(hourInput.value, "0");
+
+            changeInputThenBlur(minuteInput, "0");
+            assert.strictEqual(minuteInput.value, "0");
+
+            changeInputThenBlur(secondInput, "0");
+            assert.strictEqual(secondInput.value, "0");
+
+            changeInputThenBlur(millisecondInput, "0");
+            assert.strictEqual(millisecondInput.value, "0");
+
+            /* I can select time between default minTime and default maxTime */
+            changeInputThenBlur(hourInput, "12");
+            assert.strictEqual(hourInput.value, "12");
+
+            changeInputThenBlur(minuteInput, "30");
+            assert.strictEqual(minuteInput.value, "30");
+
+            changeInputThenBlur(secondInput, "30");
+            assert.strictEqual(secondInput.value, "30");
+
+            changeInputThenBlur(millisecondInput, "500");
+            assert.strictEqual(millisecondInput.value, "500");
+
+            /* I can select default maxTime */
+            changeInputThenBlur(hourInput, "23");
+            assert.strictEqual(hourInput.value, "23");
+
+            changeInputThenBlur(minuteInput, "59");
+            assert.strictEqual(minuteInput.value, "59");
```

Comment:
these change-assert pairs aren't helpful since the code inside `changeInputThenBlur` literally sets `input.value`, then you assert that that property has been changed. 
a) of course it has, you just changed it. 
b) DOM state is not the important thing to test here--we want to test the React component (its state and props).

would be much better to do all the DOM event manipulation, then test that the React component updated as expected: its internal state and event callbacks reflect what you expect.